### PR TITLE
fix: honor 'active' flag for ethernet interfaces

### PR DESCRIPTION
**Pull request by uskr**
There was no code to check if the ethernet interface capture was enabled or not. Added the check.

### CODE REVIEW

**Comment by uskr** [comment]
added one more commit fixing an out-of-bounds situation.

**Comment by uskr** [comment]
Added some extra commits:
- added configuration option to specify the buffer capacity (make it easier to run on embedded devices with limited memory)
- fixed a possible out of bounds

File: tap/nzyme-tap.conf.example [hunk 1]
```diff
@@ -2,6 +2,8 @@
 leader_secret = "your_key_copied_from_nzyme_web_interface"
 leader_uri = "https://nzyme.example.org:22900/"
 accept_insecure_certs = false
+wifi_pkt_buffer_capacity = 65535
+ethernet_pkt_buffer_capacity = 65535
```
Comment (lennartkoopmann) [inline tap/nzyme-tap.conf.example:6]: Two thoughts:

* I think those should go into the `performance` section of the config file
* I would rename them to `wifi_broker_buffer_capacity` and `ethernet_broker_buffer_capacity`. This way, we have more flexibility for future buffer capacity configurations (like the processor buffers for `dns` or `arp`)

What do you think?

File: tap/src/main.rs [hunk 2]
```diff
@@ -59,7 +59,7 @@ fn main() {
     ethernet::capture::print_devices();
     
     let metrics = Arc::new(Mutex::new(metrics::Metrics::new()));
-    let bus = Arc::new(Bus::new(metrics.clone(), "ethernet_packets".to_string()));
+    let bus = Arc::new(Bus::new(metrics.clone(), "ethernet_packets".to_string(), configuration.general.wifi_pkt_buffer_capacity, configuration.general.ethernet_pkt_buffer_capacity));
```
Comment (lennartkoopmann) [inline tap/src/main.rs:62]: Should we pass the entire `configuration` here? I have a feeling we'll use more config variables in the `Bus` in the future.

File: tap/src/main.rs [hunk 3]
```diff
@@ -82,7 +82,11 @@ fn main() {
     // Ethernet Capture.
     let ethernet_capture_conf = configuration.clone();
     if let Some(ethernet_interfaces) = ethernet_capture_conf.ethernet_interfaces {
-        for (interface_name, _) in ethernet_interfaces {
+        for (interface_name, interface_config) in ethernet_interfaces {
+            if !interface_config.active {
+                info!("Skipping disabled Ethernet interface [{}].", interface_name);
+                continue;
+            }
```
Comment (lennartkoopmann) [inline tap/src/main.rs:89]: Doh! Thanks! :laughing:

**Review by lennartkoopmann — CHANGES_REQUESTED** [review]
LGTM, just a few cosmetic suggestions.

File: tap/src/main.rs (hunk 2 above)
Comment (uskr) [reply tap/src/main.rs:62]: Sounds good. will change and add the commits.

File: tap/nzyme-tap.conf.example (hunk 1 above)
Comment (uskr) [reply tap/nzyme-tap.conf.example]: done.

File: tap/src/main.rs (hunk 2 above)
Comment (uskr) [reply tap/src/main.rs]: done.

**Review by lennartkoopmann — APPROVED** [review]

**Comment by lennartkoopmann** [comment]
Thank you! Added new configuration variables to changelog ticket in #870

**Comment by lennartkoopmann** [comment]
Removed to now obsolete TODO comments in 113950c

**Pull request merged**